src/components/ciConfig/TargetPlatformSelector.tsx

```diff
@@ -0,0 +1,162 @@
+    setShowCustomPlatformWarning,
+    targetPlatformMap,
+    parentState,
+}: TargetPlatformSelector) {
+    const platformMenuList = (props): JSX.Element => {
+        return (
+            <components.MenuList {...props}>
+                <div className="cn-5 pl-12 pt-4 pb-4 dc__italic-font-style">
+                    Type to enter a target platform. Press Enter to accept.
+                </div>
+                {props.children}
+            </components.MenuList>
+        )
+    }
+
+    const noMatchingPlatformOptions = (): string => {
+        return 'No matching options'
+    }
+    const handlePlatformChange = (selectedValue): void => {
+        setSelectedTargetPlatforms(selectedValue)
+    }
+
+    const platformOption = (props): JSX.Element => {
+        const { selectOption, data } = props
+        return (
+            <div
+                onClick={(e) => selectOption(data)}
+                className="flex left pl-12"
+                style={{ background: props.isFocused ? 'var(--N100)' : 'transparent' }}
+            >
+                {!data.__isNew__ && (
+                    <input
+                        checked={props.isSelected}
+                        type="checkbox"
+                        style={{ height: '16px', width: '16px', flex: '0 0 16px' }}
+                    />
+                )}
+                <div className="flex left column w-100">
+                    <components.Option className="w-100 option-label-padding" {...props} />
+                </div>
+            </div>
+        )
+    }
+
+    const handleCreatableBlur = (event): void => {
+        if (event.target.value) {
+            setSelectedTargetPlatforms([
+                ...selectedTargetPlatforms,
+                {
+                    label: event.target.value,
+                    value: event.target.value,
+                },
+            ])
+            if (!showCustomPlatformWarning) {
+                setShowCustomPlatformWarning(!targetPlatformMap.get(event.target.value))
+            }
+        } else {
+            setShowCustomPlatformWarning(
+                selectedTargetPlatforms.some((targetPlatform) => !targetPlatformMap.get(targetPlatform.value)),
+            )
+        }
+    }
+
+    const handleKeyDown = (event): void => {
+        if (event.key === 'Enter' || event.key === 'Tab') {
+            event.target.blur()
+        }
+    }
+
+    const getOverridenValue = () => {
+        // let targetPlatform =
+        //     parentState?.selectedCIPipeline?.dockerConfigOverride?.ciBuildConfig?.dockerBuildConfig?.targetPlatform
+        let targetPlatform =
+            parentState?.ciConfig?.ciBuildConfig?.dockerBuildConfig?.targetPlatform
+            if(!targetPlatform){
+             return <div className='bcn-1 br-4 flex cn-7 pt-8 pb-8'>
+                Target platform is not set
+              </div>
+            }else{
+              if (targetPlatform && !targetPlatform?.includes(',')) {
```

Comment:
no need to add check for `targetPlatform` as it is already getting checked in if

Reply:
done